common/src/main/scala/org/neo4j/spark/writer/BaseDataWriter.scala

```diff
@@ -91,15 +93,25 @@ abstract class BaseDataWriter(jobId: String,
     Unit
   }
 
+  /**
+   * df: we check if the thrown exception is STOPPED_THREAD_EXCEPTION. This is the
+   * exception that is thrown when the streaming query is interrupted, we don't want to cause
+   * any error in this case. The transaction are rolled back automatically.
+   */
   private def logAndThrowException(e: Exception): Unit = {
-    if (e.isInstanceOf[ClientException]) {
-      log.error(s"Cannot commit the transaction because: ${e.getMessage}")
+    if(e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE) {
+      logInfo(e.getMessage)
     }
     else {
-      log.error("Cannot commit the transaction because the following exception", e)
-    }
+      if (e.isInstanceOf[ClientException]) {
```

Comment:
maybe makes sense eventually change the if in this way:
```suggestion
      if ((e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE) || e.isInstanceOf[ClientException]) {
```

and remove the the if above if we cannot leverage the exception root cause